### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by leobugeja**
'Fixed' a possible issue with the prediction function. Before whichever bird had a greater than 0.5 prediction always was picked as the predicted bird (due to the rounding function). However, problem arrises if you have say 20 birds, then the largest probability bird might have a probabilty of 0.4 and the other 19 like 0.03 but that 0.4 wouldn't be picked because its not over 0.5. Therefore I've changed it to an cutoff which varies depending on the number of categories. In my example I've used 1/sqrt(num_birds) as the cuttoff, so if theres 16 birds, the max bird probability needs to be higher than 0.25 (1/sqrt(16)) to be registered. If its not registered I've put it in a seperate gategory: "Unidentified audio":
image
Above I used a cutoff of 0.7 just to demonstrate that it works.

I also added progress bars for the listen function and added an expection in listen if the dataframe is empty, for example if all the audio is filtered out or if the original audio was silent.

### CODE REVIEW

**Comment by leobugeja** [comment]
Very important addition I think

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Francis
-## Audio Recognition Neural Network
+## Audio Recognition Neural Network 🐦
```
Comment (chrishop) [inline README.md:3]: ahah love the whimsy 🦆

File: README.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ francis listen path/to/wav/file path/to/keras/model.h5
 ## (Not So) Quick Start
 
 ### Data
-First you need some data, you can collect it from your own sources or use the[xeno-canto](https://github.com/ntivirikin/xeno-canto-py) python api.
+First you need some data, you can collect it from your own sources or use the [xeno-canto](https://github.com/ntivirikin/xeno-canto-py) python api.
```
Comment (chrishop) [inline README.md:58]: yea thanks for this, I saw it im my PR but  I cba to change it at the time 😅

File: francis/__main__.py [hunk 3]
```diff
@@ -209,21 +209,30 @@ def listen(audio_sample, model_path):
...
+    preprocess_bar = default_bar(
+        "noise reducing and high pass filtering \t\t\t", len(pre_df)
+    )
     the_df = preprocess.process(
-        pre_df, CONFIG["SAMPLE_RATE"], CONFIG["PREPROCESSING_ON"]
+        pre_df,
+        CONFIG["SAMPLE_RATE"],
+        CONFIG["PREPROCESSING_ON"],
+        bar_config=preprocess_bar,
     )
```
Comment (chrishop) [inline francis/__main__.py:220]: Thanks for doing it this way, Its not the best way of doing it, but its the clearest way, while keeping seperation of logic. The downside is that it creates a little bit of clutter!

File: francis/__main__.py [hunk 4]
```diff
@@ -238,14 +247,23 @@ def listen(audio_sample, model_path):
...
+    print(around(the_model.predict(spectrograms), decimals=3))
 
     print("load categories")
     categories = io.load_categories(model_path)
 
     print("predictions ..")
-    #  print(model_adaptor.adapt_predictions(predictions, categories))
-    print(model_adaptor.summarize_predictions(predictions, categories))
+    bird_prediction = model_adaptor.predicted_bird_timestamp(predictions, categories)
+    print(model_adaptor.prediction_string_format(bird_prediction, categories))
```
Comment (chrishop) [inline francis/__main__.py:267]: thanks for splitting this up to, this makes it easier to write tests for the logic in future.

**Review by chrishop — APPROVED** [review]

**Pull request merged**